src/gui/tag/TagsEdit.cpp

```diff
@@ -337,7 +337,7 @@ struct TagsEdit::Impl
         assert(i < tags.size());
         auto occurrencesOfCurrentText =
             std::count_if(tags.cbegin(), tags.cend(), [this](const auto& tag) { return tag.text == currentText(); });
-        if (currentText().isEmpty() || occurrencesOfCurrentText > 1) {
+        if ((currentText().isEmpty() && tags.size() > 1) || occurrencesOfCurrentText > 1) {
             tags.erase(std::next(tags.begin(), std::ptrdiff_t(editing_index)));
```

Comment:
Yikes this line is what causes the crash and needs to be replaced with a safer option.

Reply:
The change fixes the specific problem. The tags list will have at least size of 1, such that even after running the function, `editing_index` will be `>=0` and the invariant hold. 

Adding guards is probably a good idea, but I'd suggest to do some refactoring instead: What would be a good fallback for currentText() if editing_index is OoB? Crash gracefully? `currentRect` and `currentText` may return a reference.

Reply:
currentText should return an empty string and currentRect a zero sized rect.

Reply:
Given the current design of the class, I'd consider handling out-of-bounds as follows:
If tags is not empty, set editing_index to 0, then return tags[editing_index].
If tags is empty, add an empty tag, set editing_index to 0 and return.

Alternatively, we could introduce set/get currentText()/currentRect() member functions to avoid the reference problem (`currentText()` and `currentRect()` return references that are actually written to https://github.com/libklein/keepassxc/blob/develop/src/gui/tag/TagsEdit.cpp#L399. This would be my preferred option.

Reply:
That's just nuts, we should never be returning a reference to a QString. QString is optimized in the backend.